File: server/src/game/conf/GameConfig.go
package conf

import (
	//"bufio"
	"encoding/json"
	"fmt"
	//"io"
	"io/ioutil"
	"os"
	//"strings"
)

type GameConfigElem struct {
	Id int
	Data string
	Des string
	
}

type GameConfig0 struct {
	Dic map[int]*GameConfigElem
	Arr []*GameConfigElem
}

var (
	GameConfig GameConfig0
)

func loadGameConfig(path string) {
	file, err := os.Open(path)
	defer func() {
		if file != nil {
			file.Close()
		}
	}()
	if err != nil {
		panic(fmt.Sprint("load config error, path=", path))
	}

	GameConfig.Dic = make(map[int]*GameConfigElem)
	GameConfig.Arr = make([]*GameConfigElem, 0, 32)

	bt,err  :=  ioutil.ReadAll(file)
	json.Unmarshal(bt, &GameConfig.Arr)
	for _,conf := range GameConfig.Arr{
		GameConfig.Dic[conf.Id] = conf
	}
	
}

File: tools/conf/output/ts/GameConfig.ts
class GameConfig {
    private static _dic:Dictionary<GameConfig> = new Dictionary<GameConfig>();
    private static _arr:Array<GameConfig> = new Array<GameConfig>();

	 public Id :number ;
     public Data :string ;
    
    

    
    public constructor(){

    }


    public static get Dic():Dictionary<GameConfig>{
        return  GameConfig._dic;
    }
    public static get Arr():Array<GameConfig>{
        return GameConfig._arr;
    }
    public static parse(json:any[]){

        for(var i = 0; i < json.length; i++){
            let config:GameConfig = <GameConfig>json[i];
            GameConfig._arr.push(config);
            GameConfig._dic.add(config.Id,config);
        }
    }
}